Index: libvdpau/src/vdpau_wrapper.c
===================================================================
--- libvdpau.orig/src/vdpau_wrapper.c	2011-08-29 20:03:58.000000000 +0200
+++ libvdpau/src/vdpau_wrapper.c	2011-08-29 20:14:42.000000000 +0200
@@ -133,6 +133,12 @@
                  "", vdpau_driver, "");
         backend_dll = dlopen(vdpau_driver_lib, RTLD_NOW | RTLD_GLOBAL);
     }
+    if (!backend_dll) {
+        /* Try again using the "legacy" /usr/lib/vdpau/ path */
+        snprintf(vdpau_driver_lib, sizeof(vdpau_driver_lib), DRIVER_LIB_FORMAT,
+                 "/usr/lib/vdpau/", vdpau_driver, ".1");
+        backend_dll = dlopen(vdpau_driver_lib, RTLD_NOW | RTLD_GLOBAL);
+    }
 
     if (vdpau_driver_dri2) {
         XFree(vdpau_driver_dri2);
